### PR TITLE
Subversion provider should respect proxies

### PR DESCRIPTION
**Pull request by chefsalim**
See issue at https://github.com/chef/chef/issues/4214

This change allows the subversion provider to respect proxy environment variables, by adding an appropriate proxy configuration to the svn command line.

@chef/client-core @randomcamel @tyler-ball

### CODE REVIEW

**Comment by tyler-ball** [comment]
:+1: Nice job!  Will you update the `proxy_tests` project to [uncomment](https://github.com/chef/proxy_tests/blob/ddbc8d67a40e79a2e4c6dad3db9544b2d3383f4b/files/default/scripts/repo/cookbooks/test/recipes/default.rb#L35-L37) the svn resource test after this is merged to master?

**Comment by chefsalim** [comment]
Yep, will do.

File: lib/chef/provider/subversion.rb [hunk 1]
```diff
@@ -194,6 +196,21 @@ def authentication
         result
       end
 
+      def proxy
+        %w{http https}.each { |p|
+          if @new_resource.repository.start_with?("#{p}:") &&
+              ENV["#{p}_proxy"] != nil &&
```
Comment (randomcamel) [inline lib/chef/provider/subversion.rb]: Don't we want to pull this from `Chef::Config` instead of `ENV`?

File: lib/chef/provider/subversion.rb [hunk 2]
```diff
@@ -194,6 +196,21 @@ def authentication
...
+              !fuzzy_hostname_match_any?(URI.parse(@new_resource.repository).host, ENV["no_proxy"])
+            proxy_uri = URI.parse(ENV["#{p}_proxy"])
```
Comment (randomcamel) [inline lib/chef/provider/subversion.rb]: `Chef::HTTP::BasicClient#proxy_uri` does different processing for the proxy URI, it'd be good to at least check out the difference.

File: lib/chef/provider/subversion.rb (hunk 1 above)
Comment (chefsalim) [inline lib/chef/provider/subversion.rb]: The Chef::Config proxy values will get exported, so we want to pick them up from env (to handle the case where chef config does not have a specified value).

Comment (randomcamel) [inline lib/chef/provider/subversion.rb]: Huh, I had it in my head it was bi-directional and Config would return ENV values, but it looks not.

**Comment by randomcamel** [comment]
:+1:

**Pull request merged**